source/routing/journey.cpp

```diff
@@ -0,0 +1,118 @@
+along with this program. If not, see <http://www.gnu.org/licenses/>.
+
+Stay tuned using
+twitter @navitia
+IRC #navitia on freenode
+https://groups.google.com/d/forum/navitia
+www.navitia.io
+*/
+
+#include "journey.h"
+#include "type/type.h"
+
+using namespace navitia::routing;
+
+Journey::Section::Section(const type::StopTime& in,
+                          const DateTime in_dt,
+                          const type::StopTime& out,
+                          const DateTime out_dt):
+    get_in_st(&in),
+    get_in_dt(in_dt),
+    get_out_st(&out),
+    get_out_dt(out_dt)
+{}
+
+bool Journey::Section::operator==(const Section & rhs) const {
+
+    return get_in_st->order() == rhs.get_in_st->order()
+        && get_in_dt == rhs.get_in_dt
+        && get_out_st->order() == rhs.get_out_st->order()
+        && get_out_dt == rhs.get_out_dt;
+}
+
+bool Journey::is_pt() const {
+    if (sections.empty()) {
+        return false;
+    } else {
+        return true;
+    }
+}
+
+bool Journey::better_on_dt(const Journey& that, bool request_clockwise) const {
+    if (request_clockwise) {
+        if (arrival_dt != that.arrival_dt) { return arrival_dt <= that.arrival_dt; }
+        if (departure_dt != that.departure_dt) { return departure_dt >= that.departure_dt; }
+    } else {
+        if (departure_dt != that.departure_dt) { return departure_dt >= that.departure_dt; }
+        if (arrival_dt != that.arrival_dt) { return arrival_dt <= that.arrival_dt; }
+    }
+    // FIXME: I don't like this objective, for me, this is a
+    // transfer objective, but then you can return some solutions
+    // that we didn't return before.
+    if (! (better_on_transfer(that, request_clockwise) &&
+           that.better_on_transfer(*this, request_clockwise))) {
+        // if they are not equal on transfer, we don't check min_waiting_dur
+        return true;
+    }
+    return min_waiting_dur >= that.min_waiting_dur;
+}
+
+bool Journey::better_on_transfer(const Journey& that, bool) const {
+    if (sections.size() != that.sections.size()) {
+        return sections.size() <= that.sections.size();
+    }
+    return nb_vj_extentions <= that.nb_vj_extentions;
+}
+
+bool Journey::better_on_sn(const Journey& that, bool) const {
+    //we consider the transfer sections also as walking sections
+    return sn_dur + transfer_dur <= that.sn_dur + that.transfer_dur;
+}
+
+bool Journey::operator==(const Journey & rhs) const {
+    return departure_dt == rhs.departure_dt
+        && arrival_dt == rhs.arrival_dt
+        && sections == rhs.sections;
+}
+
+
+size_t SectionHash::operator()(const Journey::Section& s, size_t seed) const {
+
+    boost::hash_combine(seed, s.get_in_st->order());
+    boost::hash_combine(seed, s.get_in_dt);
+    boost::hash_combine(seed, s.get_out_st->order());
+    boost::hash_combine(seed, s.get_out_dt);
+
+    return seed;
+}
+
+size_t JourneyHash::operator()(const Journey& j) const {
```

Comment:
What do you want to do here? And by that I mean what do you consider a "similar" journey.
You only compare in and out datetime and the relative position of the stoptime in the VJ without checking that it is the same vj.

Reply:
Yes, I understood that similar journeys could be done like this, I understood wrong... I'll had `boost::hash_combine(seed, s.get_in_st->vj);` and `boost::hash_combine(seed, s.get_out_st->vj);` with the same logic in the comparison operator.